src/App.test.js

```diff
@@ -1,93 +1,77 @@
-      } else {
-        return Promise.resolve(responseToken);
-      }
-    });
+  it('renders loading text in English', () => {
+    const { getByText } = render(<App locale="en" />);
+    getByText('Loading...');
   });
 
-  it('renders app component', () => {
-    const { getByText } = render(
-      <IntlProvider locale="en" messages={flattenMessages(messages['en'])}>
-        <App />
-      </IntlProvider>,
-    );
+  it('renders loading text in Spanish', () => {
+    const { getByText } = render(<App locale="es" />);
+    getByText('Cargando...');
   });
 
-  it('fetches user and display user data', async () => {
-    const tokenDecodeData = {
-      email: 'fcoronel@makingsense',
-      name: 'fede',
-      lang: 'es',
-    };
-
-    jwt_decode.mockResolvedValue(tokenDecodeData);
-
-    const { getByText } = render(
-      <IntlProvider locale="en" messages={flattenMessages(messages['en'])}>
-        <App />
-      </IntlProvider>,
-    );
-
-    await wait(() => getByText(response.data.user.Email));
-
-    const userEmail = getByText(response.data.user.Email);
+  describe('in normal backend behavior', () => {
+    const email = 'fcoronel@makingsense.com';
+
+    beforeEach(() => {
+      // prepare localStorage double
+      global.localStorage = {
+        getItem: jest.fn(),
+        setItem: jest.fn(),
+        removeItem: jest.fn(),
+      };
+
+      // prepare backend double
+      const responseOfGetUserData = {
+        data: {
+          user: {
+            Email: email,
+          },
+        },
+      };
+
+      const responseOfGetToken = {
+        data: {
+          jwtToken: 'token',
+        },
+      };
+
+      axios.get = jest.fn((url) => {
+        if (url === process.env.REACT_APP_API_URL + '/Reports/Reports/GetUserData') {
+          return Promise.resolve(responseOfGetUserData);
+        } else if (url === process.env.REACT_APP_API_URL + '/Reports/Reports/GetJwtToken') {
+          return Promise.resolve(responseOfGetToken);
+        } else {
+          return Promise.reject('Unexpected call to backend');
+        }
+      });
+    });
 
-    expect(userEmail).toBeDefined();
-    expect(axios.get).toHaveBeenCalledTimes(1);
-    expect(axios.get).toHaveBeenCalledWith(
-      process.env.REACT_APP_API_URL + '/Reports/Reports/GetUserData',
-      {
-        withCredentials: 'include',
-      },
-    );
+    it('displays user data and fetches user data and token', async () => {
+      const { getByText } = render(<App locale="en" />);
+
+      await wait(() => getByText(email));
+
+      expect(axios.get).toHaveBeenCalledTimes(2);
+      expect(axios.get).toHaveBeenCalledWith(
+        process.env.REACT_APP_API_URL + '/Reports/Reports/GetUserData',
+        {
+          withCredentials: 'include',
+        },
+      );
+      expect(axios.get).toHaveBeenCalledWith(
+        process.env.REACT_APP_API_URL + '/Reports/Reports/GetJwtToken',
+        {
+          withCredentials: 'include',
+        },
+      );
```

Comment:
Shouldn't you check email value here? To detect if something went wrong with the data you are showing.

Reply:
Mmm, in this particular case I do not think so. I mean, we are not sending the email to the backend, and the backend is _moked_.

Also, I am cleaning JwtToken related code to reintroduce it late, at that moment, we should verify different scenarios, for example when the stored token does not match Doppler's cookie, there will be a lot of tests.